src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMargin.xaml.cs

```diff
@@ -65,22 +72,33 @@ private void TargetMenuItem_OnClick(object sender, RoutedEventArgs e)
             if (e.OriginalSource is MenuItem { DataContext: TargetMenuItemViewModel viewModel })
             {
                 Logger.Log(FunctionId.InheritanceMargin_NavigateToTarget, KeyValueLogMessage.Create(LogType.UserAction));
-                _operationExecutor.Execute(
-                    new UIThreadOperationExecutionOptions(
-                        title: EditorFeaturesResources.Navigating,
-                        defaultDescription: string.Format(ServicesVSResources.Navigate_to_0, viewModel.DisplayContent),
-                        allowCancellation: true,
-                        showProgress: false),
-                    context => GoToDefinitionHelpers.TryGoToDefinition(
-                        ImmutableArray.Create(viewModel.DefinitionItem),
-                        _workspace,
-                        string.Format(EditorFeaturesResources._0_declarations, viewModel.DisplayContent),
-                        _threadingContext,
-                        _streamingFindUsagesPresenter,
-                        context.UserCancellationToken));
+
+                var token = _listener.BeginAsyncOperation(nameof(TargetMenuItem_OnClick));
+                TargetMenuItem_OnClickAsync(viewModel).CompletesAsyncOperation(token);
             }
         }
 
+        private async Task TargetMenuItem_OnClickAsync(TargetMenuItemViewModel viewModel)
+        {
+            using var context = _operationExecutor.BeginExecute(
+                title: EditorFeaturesResources.Navigating,
+                defaultDescription: string.Format(ServicesVSResources.Navigate_to_0, viewModel.DisplayContent),
+                allowCancellation: true,
+                showProgress: false);
+
+            var rehydrated = await viewModel.DefinitionItem.TryRehydrateAsync(context.UserCancellationToken).ConfigureAwait(false);
```

Comment:
Does GoToDefinitionHelpers.TryGoToDefinition below needs to be called by UI thread?

Reply:
switched here to be entirely async.